platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ConfigurationRoleUsageIntegrationTest.groovy

```diff
@@ -640,22 +640,31 @@ class ConfigurationRoleUsageIntegrationTest extends AbstractIntegrationSpec impl
         "configurations.maybeCreateConsumableUnlocked('additionalRuntimeClasspath')"    | ConfigurationRoles.CONSUMABLE | true  | "internal unlocked role-based configuration, if it doesn't already exist"
     }
 
-    def "changing usage on detached configurations does not warn"() {
+    def "changing usage #property = #change on detached configurations fails"() {
```

Comment:
uh oh. Looks like we were a little too lenient with our deprecations before. 

This test shows adding an error where we previously did not emit a deprecation warning. 

We might need to adjust the deprecation in 8.14 or address this deprecation in 10.0

Reply:
I don't think this is actually a concern.  I've split this test in several in the detached configurations region to demo the different behaviors.  Let me know what you think.